components/brave_ads/browser/ads_p3a.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/ads_p3a.h"
+
+#include <vector>
+
+#include "base/check.h"
+#include "base/json/values_util.h"
+#include "base/notreached.h"
+#include "base/strings/strcat.h"
+#include "base/time/time.h"
+#include "brave/components/p3a/brave_p3a_log_store.h"
+#include "brave/components/p3a/brave_p3a_service.h"
+#include "brave/components/p3a_utils/bucket.h"
+#include "brave/vendor/bat-native-ads/include/bat/ads/public/interfaces/ads.mojom.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/scoped_user_pref_update.h"
+
+namespace brave_ads {
+namespace p3a {
+
+namespace {
+
+constexpr char kNewTabPageEventCountDictPref[] =
+    "brave.brave_ads.p3a.ntp_event_count";
+constexpr char kViewCountKey[] = "views";
+constexpr char kClickCountKey[] = "clicks";
+constexpr char kExpireTimeKey[] = "expiry_time";
+
+constexpr int kCountBuckets[] = {0, 1, 2, 3, 8, 12, 16};
+
+constexpr base::TimeDelta kHistogramExpiryTime = base::Days(7);
+constexpr base::TimeDelta kCountExpiryTime = base::Days(30);
+
+std::string GetHistogramName(
+    const std::string& creative_instance_id,
+    const ads::mojom::NewTabPageAdEventType event_type) {
+  std::string result = "creativeInstanceId." + creative_instance_id;
+  std::string metric_event_type;
+  switch (event_type) {
+    case ads::mojom::NewTabPageAdEventType::kServed:
+      NOTREACHED();
+      return {};
+    case ads::mojom::NewTabPageAdEventType::kViewed:
+      metric_event_type = "views";
+      break;
+    case ads::mojom::NewTabPageAdEventType::kClicked:
+      metric_event_type = "clicks";
+      break;
+  }
+  return base::StrCat({brave::kCreativeMetricPrefix, creative_instance_id, ".",
+                       metric_event_type});
+}
+
+void UpdateMetricCount(PrefService* local_state,
+                       const std::string& pref_key,
+                       const std::string& creative_instance_id,
+                       const ads::mojom::NewTabPageAdEventType event_type) {
+  DCHECK(local_state);
```

Comment:
`DCHECK(mojom::IsKnownEnumValue(event_type));`

Reply:
resolved since we're no longer using the ads lib